tools/gen_agency_cert.sh

```diff
@@ -0,0 +1,323 @@
+    echo -e "\033[31m[WARN] ${content}\033[0m"
+}
+
+LOG_INFO()
+{
+    local content=${1}
+    echo -e "\033[32m[INFO] ${content}\033[0m"
+}
+
+getname() {
+    local name="$1"
+    if [ -z "$name" ]; then
+        return 0
+    fi
+    [[ "$name" =~ ^.*/$ ]] && {
+        name="${name%/*}"
+    }
+    name="${name##*/}"
+    echo "$name"
+}
+
+check_name() {
+    local name="$1"
+    local value="$2"
+    [[ "$value" =~ ^[a-zA-Z0-9._-]+$ ]] || {
+        echo "$name name [$value] invalid, it should match regex: ^[a-zA-Z0-9._-]+\$"
+        exit "${EXIT_CODE}"
+    }
+}
+
+file_must_exists() {
+    if [ ! -f "$1" ]; then
+        echo "$1 file does not exist, please check!"
+        exit "${EXIT_CODE}"
+    fi
+}
+
+dir_must_exists() {
+    if [ ! -d "$1" ]; then
+        echo "$1 DIR does not exist, please check!"
+        exit "${EXIT_CODE}"
+    fi
+}
+
+dir_must_not_exists() {
+    if [ -e "$1" ]; then
+        echo "$1 DIR exists, please clean old DIR!"
+        exit "${EXIT_CODE}"
+    fi
+}
+
+help()
+{
+    echo "${1}"
+    cat << EOF
+Usage:
+    -c <ca path>           [Required]
+    -a <agency name>       [Required]
+    -g <gm ca path>        gmcert ca key path, if generate gm node cert 
+    -h Help
+e.g: 
+    bash $0 -c nodes/cert -a newAgency
+    bash $0 -c nodes/cert -a newAgency -g nodes/gmcert
+EOF
+exit 0
+}
+
+parse_params()
+{
+    while getopts "c:a:g:h" option;do
+        case $option in
+        c) ca_path="${OPTARG}"
+            if [ ! -e "$ca_path/ca.key" ]; then LOG_WARN "$ca_path/ca.key not exist" && exit 1; fi
+            if [ ! -e "$ca_path/ca.crt" ]; then LOG_WARN "$ca_path/ca.crt not exist" && exit 1; fi
+        ;;
+        a) agency="${OPTARG}"
+            if [ -z "$agency" ]; then LOG_WARN "$agency not specified" && exit 1; fi
+        ;;
+        g) guomi_mode="yes" && gmca_path=$OPTARG
+            if [ ! -e "$gmca_path/gmca.key" ]; then LOG_WARN "$gmca_path/gmca.key not exist" && exit 1; fi
+            if [ ! -e "$gmca_path/gmca.crt" ]; then LOG_WARN "$gmca_path/gmca.crt not exist" && exit 1; fi        
+        ;;
+        h) help;;
+        esac
+    done
+}
+
+gen_agency_cert() {
+    local chain="${1}"
+    local agencypath="${2}"
+    name=$(basename "$agencypath")
+
+    dir_must_exists "$chain"
+    file_must_exists "$chain/ca.key"
+    check_name agency "$name"
+    agencydir=$agencypath
+    dir_must_not_exists "$agencydir"
+    mkdir -p $agencydir
+
+    openssl genrsa -out $agencydir/agency.key 2048 2> /dev/null
+    openssl req -new -sha256 -subj "/CN=$name/O=fisco-bcos/OU=agency" -key "$agencydir/agency.key" -config "$chain/cert.cnf" -out "$agencydir/agency.csr" 2> /dev/null
+    openssl x509 -req -days 3650 -sha256 -CA "$chain/ca.crt" -CAkey "$chain/ca.key" -CAcreateserial\
+        -in "$agencydir/agency.csr" -out "$agencydir/agency.crt"  -extensions v4_req -extfile "$chain/cert.cnf" 2> /dev/null
+    
+    cp $chain/ca.crt $chain/cert.cnf $agencydir/
+    rm -f $agencydir/agency.csr
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5331982)